samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -0,0 +1,307 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2018 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  tiled-convolution.cpp
+ *
+ *  Description:
+ *    Sample code that illustrates how to divide data into tiles and lauch
+ *    separate kerenl per tile by using range accessors and parallel for with
+ *    offset in SYCL.
+ *
+ **************************************************************************/
+#include <SYCL/sycl.hpp>
+#include <chrono>
+#include <ctime>
+#include <iostream>
+
+// configuring the openCL local size
+template <typename index_t>
+struct opencl_configuration_t {
+  static constexpr index_t local_size_m = 1;
+  static constexpr index_t local_size_n = 32;
+};
+
+// define the 2-d matrix size
+template <typename index_t>
+class matrix_size_t {
+ public:
+  const index_t m;
+  const index_t n;
+  matrix_size_t(const index_t m_, const index_t n_) : m(m_), n(n_) {}
+  inline index_t size() const { return (m * n); }
+};
+
+// the tiled based convolution functor
```

Comment:
can you add a comment explaining how the convolution algorithm works? If you have a separate README somewhere  or an image then you can just add that to the pull request to help understand the algorithm